### DISCUSSION TITLE
Basic Authentication module for client tests

### DISCUSSION THREAD

**Discussion by wstrange** [serverpod/serverpod#721] [Ideas]
I'd like to be able to write api tests in my _client package that call authenticated endpoints.

Something like this:

```dart
  test('Basic Api test', () async {
   var client = Client('http://localhost:8080/');

    var events = await client.event.getEventsForUser(userId: 1);
    expect(events.length, equals(1));
    print('events == $events');
  });

```

The challenge is that the client is not authenticated, and the api call will fail because the endpoint is protected. 

Is there a trick or an easy way to authenticate the client in a test?  Perhaps a basic auth module?

**Comment by wstrange** [comment]
On further thought, I can see this also being useful to create cli scripts that update or maintain the database.

**Comment by fischerscode** [comment]
Maybe my custom auth example can help you. #705

**Comment by vlidholt** [comment]
This is a valid point, we should make it more clear how to do this and have a good testing flow set up out of the box.
